plugin/pkg/admission/podresourceallocation/admission.go

```diff
@@ -0,0 +1,181 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package podresourceallocation
```

Comment:
a general comment: in the future it's better to separate CRs into small pieces incrementally, like the new feature gate, the new resourceAllocated & policy fields, node authorizer & admission changes.  This way it will be easier for review and make your PR goes through faster :)

Reply:
Thanks for the review . This is the way Tim recommended that send PRs in upstream code - it makes it easier for API folks to review their part (like this PR) in one shot - and leave implementation to the area SMEs - but they also want the API PR when implementation is close to ready for release reasons - they dont want to be in a situation where API is in but implementation misses the release deadline and they have to do a lot of work to undo the API.

I'll do PRs for implementation in smaller chunks. I think we should merge this as the API is very unlikely to change at this point given that a lot of eyes looked at it in the community, and in our case we will have this going in even if the test coverage comes in a bit later. I'll rebase and update the PR.